pygmt/tests/test_session_management.py

```diff
@@ -20,3 +20,35 @@ def test_begin_end():
     begin()  # Restart the global session
     assert os.path.exists("pygmt-session.pdf")
     os.remove("pygmt-session.pdf")
+
+
+def test_gmt_compat_6_is_applied(capsys):
+    """
+    Ensure that users with old gmt.conf files won't get pygmt-session [ERROR]:
+    GMT_COMPATIBILITY: Expects values from 6 to 6; reset to 6.
+    """
+    end()  # Kill the global session
+    try:
+        with Session() as lib:
+            # pretend that gmt.conf has GMT_COMPATIBILITY = 5
+            lib.call_module("gmtset", "GMT_COMPATIBILITY 5")
+        begin()
+        with Session() as lib:
+            lib.call_module("basemap", "-R10/70/-3/8 -JX4i/3i -Ba")
+            out, err = capsys.readouterr()  # capture stdout and stderr
+            assert out == ""
+            assert err != (
```

Comment:
Note: To future maintainers on what `capsys` is. It's a pytest fixture for capturing stdout/stderr. See https://docs.pytest.org/en/5.4.2/capture.html#accessing-captured-output-from-a-test-function